app/assets/js/hooks/use-wrap-action-with-verify-dialog.js

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+import useIntroduction from './use-introduction';
+
+export default function useWrapActionWithVerifyDialog( {
+	action,
+	doNotShowAgainKey = null,
+} ) {
+	const {
+		isViewed: shouldNotShowAgain,
+		markAsViewed: markAsShouldNotShowAgain,
+	} = useIntroduction( doNotShowAgainKey );
+
+	const [
+		dialogState,
+		setDialogState,
+	] = useState( {
+		isOpen: false,
+		actionArgs: [],
+	} );
+
+	const [
+		doNotShowAgainCheckboxState,
+		setDoNotShowAgainCheckboxState,
+	] = useState( false );
+
+	return {
+		checkbox: {
+			isChecked: doNotShowAgainCheckboxState,
+			setIsChecked: setDoNotShowAgainCheckboxState,
+		},
+		dialog: {
+			isOpen: dialogState.isOpen,
+			approve: () => {
+				action( ...dialogState.actionArgs );
+
+				if ( doNotShowAgainCheckboxState && doNotShowAgainKey ) {
```

Comment:
why do you need to check the key? isn't it checked in the `useIntroduction` hook?

Reply:
key can be undefined, if I want confirmation modal without the logic of "do not show again".